x-pack/plugins/data_usage/server/services/autoops_api.ts

```diff
@@ -0,0 +1,178 @@
+              error.response.status
+            )}} ${JSON.stringify(error.response.data)}} ${requestConfigDebugStatus}`,
+            {
+              ...errorMetadataWithRequestConfig,
+              http: {
+                ...errorMetadataWithRequestConfig.http,
+                response: {
+                  status_code: error.response.status,
+                  body: error.response.data,
+                },
+              },
+            }
+          );
+          throw new Error(
+            withRequestIdMessage(`the AutoOps API could not create the autoops agent`)
+          );
+        } else if (error.request) {
+          // The request was made but no response was received
+          logger.error(
+            `[AutoOps API] Creating autoops agent failed while sending the request to the AutoOps API: ${errorLogCodeCause} ${requestConfigDebugStatus}`,
+            errorMetadataWithRequestConfig
+          );
+          throw new Error(withRequestIdMessage(`no response received from the AutoOps API`));
+        } else {
+          // Something happened in setting up the request that triggered an Error
+          logger.error(
+            `[AutoOps API] Creating autoops agent failed to be created ${errorLogCodeCause} ${requestConfigDebugStatus}`,
+            errorMetadataWithRequestConfig
+          );
+          throw new Error(
+            withRequestIdMessage('the AutoOps API could not create the autoops agent')
+          );
+        }
+      }
+    );
+
+    logger.debug(`[AutoOps API] Created an autoops agent ${response}`);
+    return response;
+  }
+
+  private createTlsConfig(autoopsConfig: AutoOpsConfig | undefined) {
+    return new SslConfig(
+      sslSchema.validate({
+        enabled: true,
+        certificate: autoopsConfig?.api?.tls?.certificate,
+        key: autoopsConfig?.api?.tls?.key,
+        certificateAuthorities: autoopsConfig?.api?.tls?.ca,
+      })
+    );
+  }
+
+  private createRequestConfigDebug(requestConfig: AxiosRequestConfig<any>) {
+    return JSON.stringify({
+      ...requestConfig,
+      data: {
+        ...requestConfig.data,
+        fleet_token: '[REDACTED]',
+      },
+      httpsAgent: {
+        ...requestConfig.httpsAgent,
+        options: {
+          ...requestConfig.httpsAgent.options,
+          cert: requestConfig.httpsAgent.options.cert ? 'REDACTED' : undefined,
+          key: requestConfig.httpsAgent.options.key ? 'REDACTED' : undefined,
+          ca: requestConfig.httpsAgent.options.ca ? 'REDACTED' : undefined,
+        },
+      },
+    });
+  }
+
+  private convertCauseErrorsToString = (error: AxiosError) => {
+    if (error.cause instanceof AggregateError) {
+      return error.cause.errors.map((e: Error) => e.message);
+    }
+    return error.cause;
+  };
+}
+
+export const autoopsApiService = new AutoOpsAPIService();
```

Comment:
Not sure it matters, but should we instantiate this in the plugin setup after starting appContextService?